plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/AbstractPinotIntegrationSmokeTest.java

```diff
@@ -1765,4 +1850,93 @@ public void testLimitAndOffsetWithPushedDownAggregates()
+                "  (SUM(metric_col1) - SUM(metric_col2)) / SUM(metric_col1) AS ratio_metric_col" +
+                "  FROM duplicate_values_in_columns WHERE dim_col != another_dim_col" +
+                "  GROUP BY string_col" +
+                "  ORDER BY string_col\""))
+                .matches("VALUES (VARCHAR 'string2', DOUBLE '1000.0', 1, DOUBLE '-1.0')");
+
+        assertThat(query("SELECT DISTINCT string_col, another_string_col" +
+                "  FROM \"SELECT string_col, another_string_col" +
+                "  FROM duplicate_values_in_columns WHERE dim_col = another_dim_col\""))
+                .matches("VALUES (VARCHAR 'string1', VARCHAR 'string1')," +
+                        "  (VARCHAR 'string1', VARCHAR 'another_string1')," +
+                        "  (VARCHAR 'string2', VARCHAR 'another_string2')");
+
+        assertThat(query("SELECT string_col, sum_metric_col1" +
+                "  FROM \"SELECT string_col," +
+                "  SUM(CASE WHEN dim_col = another_dim_col THEN metric_col1 ELSE 0 END) AS sum_metric_col1" +
+                "  FROM duplicate_values_in_columns GROUP BY string_col ORDER BY string_col\""))
+                .matches("VALUES (VARCHAR 'string1', DOUBLE '1110.0')," +
+                        "  (VARCHAR 'string2', DOUBLE '100.0')");
+
+        assertThat(query("SELECT \"percentile(int_col, 90.0)\"" +
+                "  FROM \"SELECT percentile(int_col, 90) FROM " + ALL_TYPES_TABLE + "\""))
+                .matches("VALUES (DOUBLE '56.0')");
+
+        assertThat(query("SELECT bool_col, \"percentile(int_col, 90.0)\"" +
+                "  FROM \"SELECT bool_col, percentile(int_col, 90) FROM " + ALL_TYPES_TABLE + " GROUP BY bool_col\""))
+                .matches("VALUES (true, DOUBLE '56.0')," +
+                        "  (false, DOUBLE '0.0')");
+
+        assertThat(query("SELECT \"sqrt(percentile(sqrt(int_col),'26.457513110645905'))\"" +
+                "  FROM \"SELECT sqrt(percentile(sqrt(int_col), sqrt(700))) FROM " + ALL_TYPES_TABLE + "\""))
+                .matches("VALUES (DOUBLE '2.7108060108295344')");
+
+        assertThat(query("SELECT int_col, \"sqrt(percentile(sqrt(int_col),'26.457513110645905'))\"" +
+                "  FROM \"SELECT int_col, sqrt(percentile(sqrt(int_col), sqrt(700))) FROM " + ALL_TYPES_TABLE + " GROUP BY int_col\""))
+                .matches("VALUES (54, DOUBLE '2.7108060108295344')," +
+                        "  (55, DOUBLE '2.7232698153315003')," +
+                        "  (56, DOUBLE '2.7355647997347607')," +
+                        "  (0, DOUBLE '0.0')");
+    }
+
+    @Test
+    public void testAggregationPushdownWithArrays()
+    {
+        assertThat(query("SELECT string_array_col, count(*) FROM " + ALL_TYPES_TABLE + " WHERE int_col = 54 GROUP BY 1"))
+                .isNotFullyPushedDown(ExchangeNode.class, ProjectNode.class, AggregationNode.class, ExchangeNode.class, ExchangeNode.class, AggregationNode.class, ProjectNode.class)
```

Comment:
IIRC `isNotFullyPushedDown` verifies results with and without pushdown (as long as pushdown respects `allow_pushdown_into_connectors` session config).

So maybe the `.matches` is redundant here.